tests/utils.py

```diff
@@ -256,14 +256,20 @@ def assert_guest_os_memory_amount(vm, spec_memory_amount):
     )
 
 
-def assert_restart_required_condition(vm, expected_message):
-    def _get_restart_required_condition(vm):
-        return [condition for condition in vm.instance.status.conditions if condition.type == "RestartRequired"][0]
+def get_restart_required_condition(vm):
+    conditions = [
+        condition
+        for condition in vm.instance.status.conditions
+        if condition.type == "RestartRequired" and condition.status == "True"
+    ]
+    return conditions[0] if conditions else []
 
+
+def assert_restart_required_condition(vm, expected_message):
```

Comment:
this whole code can be replaced with:
```
vm.wait_for_condition(condition="RestartRequired", status=vm.Condition.Status.TRUE....)
```

Reply:
so use `ocp_resources.resource.Resource.get_condition_message` which gets `condition_type` and `condition_status` and returns the message